Okta documentation. (#25580)
gravitational/teleport: docs/pages/application-access/okta/introduction.mdx

---
title: Okta integration with Application Access
description: How to set up Okta integration with Teleport Application Access.
---

Teleport allows applications housed within Okta to be integrated with the
Application Access service. This will result in Okta applications showing up
in Teleport's applications list in the UI and CLI, and it will further allow
for the user of Teleport's role based access control to control access within

> Okta.

Reviewer: This is the kind of helpful context that I think is missing in the guide above. _Now_ I understand the value here.

Reviewer 2: TBH I don't know if we need a separate "introduction" page. I would do this:

- Take this paragraph and integrate it at the beginning of the guide above so it's clear what you get from setting up Okta integration.
- Everything below seems more of an "architecture" so I would rename what remains of this page into "Architecture".

Users will be able request and be granted access to applications and groups
within Okta by requesting access to them within Teleport.

## How does the Okta integration work?

> Teleport provides an Okta service that is responsible for dealing with all

Reviewer 3: If the Okta service is a first-class Teleport service in the way the Application Service, SSH Service, Discovery Service, etc. are, I think we should capitalize the name of the service to be consistent with other Teleport services.

interactions with Okta. There are three main components of the Okta service:

* Synchronization
* User access

> * Access requests

Reviewer 3: ```suggestion
* Access Requests
```

> 

Reviewer 3: If you have time to make one, I think this would benefit from a Mermaid diagram, e.g. (but accurate):

```mermaid
flowchart LR
  oktaserv[Okta Service]-- "Synchronize Okta apps and user groups" --> okta[Okta API]
  proxy[Proxy Service] -- "User Traffic" -->oktaserv
  auth[Auth Service]-- "Validate roles and Access Requests" -->oktaserv
    Users-- "Access Applications" -->proxy
```

### Synchronization

> Teleport's Okta integration uses an Okta API token and endpoint in order to interact
> with Okta's API. Teleport will then synchronize Okta's applcations and user groups
> with Teleport. Labels for these groups and applications can be customized through the

Reviewer 3: ```suggestion
Teleport's Okta integration uses an Okta API token and endpoint in order to synchronize applications and user groups
with the Okta API. Labels for these groups and applications can be customized through the
```
Suggested to condense this a bit. Are there other ways the Okta Service interacts with the Okta API?

> use of Okta Import Rule objects within Teleport.

Reviewer 3: I think it would make sense to describe Okta Import Rules in a separate paragraph with another sentence or two. They seem important to this feature, but (for me, at least) it was pretty easy to miss this detail at the end of this pagraph.

### User access

Upon logging into Teleport, users will be granted access to the Okta applications
and user groups that they have access to within Teleport. If a user has RBAC access to
an application or user group, it will be reflected within Okta.

> ### Access requests

Reviewer 3: ```suggestion
### Access Requests
```
We capitalize this other places in the docs

> Users can use Teleport's native access request functionality in order to

Reviewer: ```suggestion
Users can use Teleport's Just-in-time Access Requests in order to
```
